TIP-0002/README.md

```diff
@@ -0,0 +1,99 @@
+<pre>
+TIP: 2
+Title: Topl Address Encoding
+Author: Edmundo López Bóbeda <e.lopez@topl.me>
+Status: Draft
+Type: Standard
+Created: 2023-04-26
+License: CC-BY-4.0
+</pre>
+
+## Abstract
+
+This TIP proposes an address serialization format for the TOPL blockchain
+
+## Motivation
+
+Implementors need to understand the format of the addresses and how to encode and decode them to call different services in the node.
+
+## Specification
+
+### Introduction
+
+In the Topl ecosystem, a lock address is a data structure that stores the evidence of a Quivr proposition. Besides the evidence, the lock address data structure stores the network and ledger identifier. This extra information is useful for improving usability and tooling. We bundle together the aforementioned information and then encode into a human readable string. The end user commonly perceives this string as “the address”.
+
+### Binary Format
+
+We encode the information in the address in an array of bytes in the following form:
+
+
+
+```
+  4 bytes       4 bytes      32 bytes
+<-----------> <---------> <-------------> 
+┌────────────┬───────────┬───────────────┐
+│ network id │ ledger id │ evidence      │
+└────────────┴───────────┴───────────────┘
+```
+
+
+
+We reserved the following values for the standard Topl Networks in the first 4 bytes as the network identifier.
+
+| **Network Name**      | **Byte Values** | **Address prefix** |
+| --------------------- | --------------- | ------------------ |
+| Main Network          | `0x8A11054C`    | `mtet`             |
+| Valhalla Test Network | `0xA5BF4108`    | `vtet`             |
+| Private Network       | `0x934B1900`    | `ptet`             |
+
+When encoded, the addresses will feature the prefix in the column labeled **Address prefix**.
+
+The next four bytes are the ledger identifier. We reserve the following value for the main ledger.
+
+| **Ledger Name** | **Byte Values** | **Remarks**                                                  |
+| --------------- | --------------- | ------------------------------------------------------------ |
+| Main Ledger     | `0xE7B07A00`    | This number has the nice property of being encoded to “main” when encoding in the main net and thus forcing the prefix to be `mtetmain` in main net addresses. However, it does not work that way for the other networks prefixes. |
+

```

Comment:
I would encourage you all to think about designating a second ledger for consensus registration. That was the intent of introducing ledger as a top-level construct for addresses. You might also think about including a  ledger so that off chain DAML contracts can be identified.

Reply:
Ok, adding a new ledger is not an issue. Will discuss with the team.

Reply:
A new task was created to analyze and include all of @scasplte2 remarks into the next iteration of the tip. The task is TSDK-486